sway-core/src/semantic_analysis/namespace/root.rs

```diff
@@ -64,34 +64,58 @@ impl Root {
         })
     }
 
+    pub(crate) fn resolve_type_with_self(
+        &mut self,
+        mut type_id: TypeId,
+        self_type: TypeId,
+        span: &Span,
+        enforce_type_arguments: EnforceTypeArguments,
+        type_info_prefix: Option<&Path>,
+        mod_path: &Path,
+    ) -> CompileResult<TypeId> {
+        type_id.replace_self_type(self_type);
+        self.resolve_type(
+            type_id,
+            span,
+            enforce_type_arguments,
+            type_info_prefix,
+            mod_path,
+        )
+    }
+
     pub(crate) fn resolve_type(
```

Comment:
Perhaps this could be a good time to add some docs for `resolve_type` and `resolve_type_with_self`, particularly as we're adding a new `type_info_prefix` arg?

Reply:
I added this to this issue: https://github.com/FuelLabs/sway/issues/1673